org.adempiere.base/src/org/compiere/model/AttachmentDBLOB.java

```diff
@@ -66,36 +68,50 @@ public class AttachmentDBLOB implements IAttachmentStore
+                return true;
+    	}
 
-        return loadXML(attach, data);
+        return loadEntries(attach, data);
     }
 
     /**
-     * Load attachment entries from XML
+     * Load attachment entries from XML or from AD_AttachmentFile
      * @param attach attachment record
      * @param data xml meta data
      * @return true if success, false otherwise
      */
-    private boolean loadXML(MAttachment attach, byte[] data) {
-        NodeList entries = getEntriesFromXML(data);
-        for (int i = 0; i < entries.getLength(); i++) {
-            final Node entryNode = entries.item(i);
-            final NamedNodeMap attributes = entryNode.getAttributes();
-            final Node nameNode = attributes.getNamedItem("name");
-            final Node sizeNode = attributes.getNamedItem("size");
-            if(nameNode==null) {
-                log.severe("No filename for entry " + i);
-                attach.m_items = null;
-                return false;
+    private boolean loadEntries(MAttachment attach, byte[] data) {
+    	if (MAttachment.LIST_IN_ATTACHMENT_FILE.equals(attach.getTitle())) {
+			int index = 0;
+    		for (MAttachmentFile attachFile : attach.getAttachmentFiles()) {
+				IAttachmentLazyDataSource ds = new AttachmentDBLazyDataSource(attach.getAD_Attachment_ID(), index, attachFile.getFileName(), Long.toString(attachFile.getFileSize().longValue()));
+				final MAttachmentEntry entry = new MAttachmentEntry(attachFile.getFileName(), attach.m_items.size() + 1, ds);
+				entry.setSHA256Sum(attachFile.getSHA256Checksum());
+				attach.m_items.add(entry);
+				index++;
+    		}
+    	} else {
+    		// XML
+            NodeList entries = getEntriesFromXML(data);
+            for (int i = 0; i < entries.getLength(); i++) {
+                final Node entryNode = entries.item(i);
+                final NamedNodeMap attributes = entryNode.getAttributes();
+                final Node nameNode = attributes.getNamedItem("name");
+                final Node sizeNode = attributes.getNamedItem("size");
+                if(nameNode==null) {
+                    log.severe("No filename for entry " + i);
+                    attach.m_items = null;
+                    return false;
+                }
+                IAttachmentLazyDataSource ds = new AttachmentDBLazyDataSource(attach.getAD_Attachment_ID(), i, nameNode.getNodeValue(), sizeNode.getNodeValue());
+                final MAttachmentEntry entry = new MAttachmentEntry(nameNode.getNodeValue(), attach.m_items.size() + 1, ds);
+                attach.m_items.add(entry);
             }
-            IAttachmentLazyDataSource ds = new AttachmentDBLazyDataSource(attach.getAD_Attachment_ID(), i, nameNode.getNodeValue(), sizeNode.getNodeValue());
-            final MAttachmentEntry entry = new MAttachmentEntry(nameNode.getNodeValue(), attach.m_items.size() + 1, ds);
-            attach.m_items.add(entry);
-        }
+    	}
         return true;
```

Comment:
It probably looks a lot cleaner to implement the ad_attachmentfile loading code in a new method and leave the existing loadxml method as it is. Same goes for the save method.

Anyway, this is more of my personal preference (and to minimize changes to existing code), it is fine for me if you prefer the current approach.